src/modules/cpu/kernel/spectrogram.hpp

```diff
@@ -0,0 +1,394 @@
+                    float real = 0.0f;
+                    float imag = 0.0f;
+                    __m256 pReal, pImag;
+                    pReal = avx_p0;
+                    pImag = avx_p0;
+                    int i = 0;
+                    for (; i < alignednfftLength; i += 8)
+                    {
+                        __m256 pSrc, pSin, pCos;
+                        pSrc = _mm256_loadu_ps(windowOutputTemp);
+                        pCos = _mm256_loadu_ps(cosfTemp);
+                        pSin = _mm256_loadu_ps(sinfTemp);
+                        pReal = _mm256_add_ps(pReal, _mm256_mul_ps(pSrc, pCos));
+                        pImag = _mm256_add_ps(pImag, _mm256_mul_ps(_mm256_mul_ps(pSrc, avx_pm1), pSin));
+                        windowOutputTemp += 8;
+                        cosfTemp += 8;
+                        sinfTemp += 8;
+                    }
+                    real = reduce_add_ps1(pReal);
+                    imag = reduce_add_ps1(pImag);
+                    for (; i < nfft; i++)
+                    {
+                        float x = *windowOutputTemp++;
+                        real += x * *cosfTemp++;
+                        imag += -x * *sinfTemp++;
+                    }
+                    fftReal[k] = real;
+                    fftImag[k] = imag;
+                }
+
+                Rpp32f *fftRealTemp = fftReal;
+                Rpp32f *fftImagTemp = fftImag;
+                int i = 0;
+                for (; i < alignednbinsLength; i += 8)
+                {
+                    __m256 pReal, pImag, pTotal;
+                    pReal = _mm256_loadu_ps(fftRealTemp);
+                    pImag = _mm256_loadu_ps(fftImagTemp);
+                    pReal = _mm256_mul_ps(pReal, pReal);
+                    pImag = _mm256_mul_ps(pImag, pImag);
+                    pTotal = _mm256_add_ps(pReal, pImag);
+                    if (power == 1)
+                    {
+                        pTotal = _mm256_sqrt_ps(pTotal);
+                    }
+                    float *pTotalPtr = (float *)&pTotal;
+                    for (Rpp32s j = i; j < (i + 8); j++)
+                    {
+                        dstPtrTemp[j * hStride + w] = pTotalPtr[j - i];
+                    }
+                    fftRealTemp += 8;
+                    fftImagTemp += 8;
+                }
+                for (; i < numBins; i++)
+                {
+                    float real = *fftRealTemp++;
+                    float imag = *fftImagTemp++;
+                    float total = (real * real) + (imag * imag);
+                    int64_t outIdx = (i * hStride + w);
+                    if (power == 1)
+                    {
+                        total = std::sqrt(total);
+                    }
+                    dstPtrTemp[outIdx] = total;
+                }
+            }
+            free(fftReal);
+            free(fftImag);
+            free(windowOutput);
+        }
+    }
+    else
```

Comment:
Most of the code below in else is same as the if. Why do we need two sets? Pls try to consolidate into one by just separating the code differences as small inline helpers

Reply:
Done